yt_dlp/extractor/tarangplus.py

```diff
@@ -0,0 +1,202 @@
+    get_elements_html_by_class,
+    traverse_obj,
+    urljoin,
+)
+
+
+class TarangPlusBaseIE(InfoExtractor):
+    _BASE_URL = 'https://tarangplus.in'
+
+
+class TarangPlusVideoIE(TarangPlusBaseIE):
+    IE_NAME = 'tarangplus:video'
+    _VALID_URL = r'https?://(?:www\.)?tarangplus\.in/(?:movies|[^#?/]+/[^#?/]+)/(?!episodes$)(?P<id>[^#?/]+)'
+    _TESTS = [{
+        'url': 'https://tarangplus.in/tarangaplus-originals/khitpit/khitpit-ep-10',
+        'md5': '78ce056cee755687b8a48199909ecf53',
+        'info_dict': {
+            'id': '67b8206719521d054c0059b7',
+            'display_id': 'khitpit-ep-10',
+            'ext': 'mp4',
+            'title': 'Khitpit Ep-10',
+            'description': 'md5:a45b805cb628e15c853d78b0406eab48',
+            'thumbnail': r're:https?://.*\.jpg',
+            'duration': 756.0,
+            'timestamp': 1740355200,
+            'upload_date': '20250224',
+            'media_type': 'episode',
+            'categories': ['Originals'],
+        },
+    }, {
+        'url': 'https://tarangplus.in/tarang-serials/bada-bohu/bada-bohu-ep-233',
+        'md5': 'b4f9beb15172559bb362203b4f48382e',
+        'info_dict': {
+            'id': '680b9d6c19521d054c007782',
+            'display_id': 'bada-bohu-ep-233',
+            'ext': 'mp4',
+            'title': 'Bada Bohu  | Ep -233',
+            'description': 'md5:e6b8e7edc9e60b92c1b390f8789ecd69',
+            'thumbnail': r're:https?://.*\.jpg',
+            'duration': 1392.0,
+            'timestamp': 1745539200,
+            'upload_date': '20250425',
+            'media_type': 'episode',
+            'categories': ['Prime'],
+        },
+    }, {
+        'url': 'https://tarangplus.in/short/ai-maa/ai-maa',
+        'only_matching': True,
+    }, {
+        'url': 'https://tarangplus.in/shows/tarang-cine-utsav-2024/tarang-cine-utsav-2024-seg-1',
+        'only_matching': True,
+    }, {
+        'url': 'https://tarangplus.in/music-videos/chori-chori-bohu-chori-songs/nijara-laguchu-dhire-dhire',
+        'only_matching': True,
+    }, {
+        'url': 'https://tarangplus.in/kids-shows/chhota-jaga/chhota-jaga-ep-33-jamidar-ra-khajana-adaya',
+        'only_matching': True,
+    }, {
+        'url': 'https://tarangplus.in/movies/swayambara',
+        'only_matching': True,
+    }]
+
+    def decrypt(self, data, key):
+        if not Cryptodome.AES:
+            raise ExtractorError('pycryptodomex not found. Please install', expected=True)
+        iv = binascii.unhexlify('00000000000000000000000000000000')
+        cipher = Cryptodome.AES.new(base64.b64decode(key), Cryptodome.AES.MODE_CBC, iv)
+        return cipher.decrypt(base64.b64decode(data)).decode('utf-8')
+
+    def _real_extract(self, url):
+        display_id = self._match_id(url)
+        webpage = self._download_webpage(url, display_id)
+        hidden_inputs_data = self._hidden_inputs(webpage)
+        json_ld_data = self._search_json_ld(webpage, display_id)
+        del json_ld_data['url']
```

Comment:
Let's do this non-fatally
```suggestion
        json_ld_data = self._search_json_ld(webpage, display_id)
        json_ld_data.pop('url', None)

```